From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Joyee Cheung <joyeec9h3@gmail.com>
Date: Tue, 2 Jun 2026 16:33:56 +0200
Subject: src: fix external reference registration for new CFunction lifetime
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Refs: https://chromium-review.googlesource.com/c/v8/v8/+/7828135
Signed-off-by: Joyee Cheung <joyeec9h3@gmail.com>
PR-URL: https://github.com/nodejs/node/pull/63679
Reviewed-By: René <contact.9a5d6388@renegade334.me.uk>
Reviewed-By: Anna Henningsen <anna@addaleax.net>
(cherry picked from commit 0401a2d083bbfa2a816cb2906597a3df7d69b20a)

diff --git a/src/node_wasi.cc b/src/node_wasi.cc
index 7d8d056620c4c6d8692646273e822def1aa21156..6c2adc826571daf07798c9aed941a0c8e11e89e2 100644
--- a/src/node_wasi.cc
+++ b/src/node_wasi.cc
@@ -234,7 +234,7 @@ void WASI::New(const FunctionCallbackInfo<Value>& args) {
 template <typename FT, FT F, typename R, typename... Args>
 void WASI::WasiFunction<FT, F, R, Args...>::SetFunction(
     Environment* env, const char* name, Local<FunctionTemplate> tmpl) {
-  auto c_function = CFunction::Make(FastCallback);
+  static auto c_function = CFunction::Make(FastCallback);
   Local<FunctionTemplate> t =
       FunctionTemplate::New(env->isolate(),
                             SlowCallback,
